lib/db/instance.go

```diff
@@ -48,9 +48,7 @@ func (db *instance) updateFiles(folder, device []byte, fs []protocol.FileInfo, m
 			err = ef.Unmarshal(bs)
 		}
 
-		// Local flags or the invalid bit might change without the version
-		// being bumped. The IsInvalid() method handles both.
-		if err == nil && ef.Version.Equal(f.Version) && ef.IsInvalid() == f.IsInvalid() {
+		if unchanged(f, ef, err == nil) {
```

Comment:
Passing the bool is weird. `err == nil && dbEquivalent(f, ef)` scans better imho